### PR TITLE
Perform Crash delegate callbacks on main thread

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* [X] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Asd per the [documentation](https://docs.microsoft.com/en-us/appcenter/sdk/crashes/macos#get-information-about-the-sending-status-for-a-crash-log), delegate methods should be performed on UI thread: 

## Related PRs or issues

#1723 

List related PRs and other issues.

## Misc

[AB#66547](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/66547)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,10 @@
 
 * **[Fix]** Fix crash involving SDK's `ms_viewWillAppear` method.
 
+### App Center Crashes
+
+* **[Fix]** Fix `MSCrashesDelegate` callback methods not being called on main thread
```
Comment (achocron) [inline CHANGELOG.md]:
```suggestion
* **[Fix]** Fix `MSCrashesDelegate` callback methods not being called on main thread.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -13,6 +13,10 @@
 
 * **[Fix]** Fix crash involving SDK's `ms_viewWillAppear` method.
 
+### App Center Crashes
+
+* **[Fix]** Fix `MSCrashesDelegate` callback methods not being called on the main thread.
```
Comment (jaelim-ms) [inline CHANGELOG.md]: @achocron We want to mention this as a behavior change, right?

Comment (achocron) [reply CHANGELOG.md]:
```suggestion
* **[Behavior Change]** Fix `MSCrashesDelegate` callback methods not being called on the main thread.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,10 @@
 
 * **[Fix]** Fix crash involving SDK's `ms_viewWillAppear` method.
 
+### App Center Crashes
+
+* **[Behavior Change]** `MSCrashesDelegate` callback methods are now invoked on the main thread (`willSendErrorReport`,  `didSucceedSendingErrorReport`, and `didFailSendingErrorReport`).
```
Comment (guperrot) [inline CHANGELOG.md]: inconsistent casing of Change vs change in this file.

Comment (jaelim-ms) [reply CHANGELOG.md]:
```suggestion
* **[Behavior change]** `MSCrashesDelegate` callback methods are now invoked on the main thread (`willSendErrorReport:`,  `didSucceedSendingErrorReport:`, and `didFailSendingErrorReport:`).
```

**Pull request merged**